examples/colormap_normalization/colormap_normalizations_piecewisenorm.py

```diff
@@ -0,0 +1,103 @@
+"""
+============================================
+Examples of arbitrary colormap normalization
+============================================
+
+Here I plot an image array with data spanning for a large dynamic range,
+using different normalizations. Look at how each of them enhances
+different features.
+
+"""
+
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
+
+from mpl_toolkits.mplot3d import Axes3D
+
+import numpy as np
+
+from sampledata import PiecewiseNormData
+
+X, Y, data = PiecewiseNormData()
+cmap = cm.spectral
+
+# Creating functions for plotting
+
+
+def make_plot(norm, label=''):
```

Comment:
Oh, I was not aware that the cbook module could be used for that purpose as well! In principle the same data would be useful for multiple examples, and even more normalization classes in the future. The reason I wanted to have it not directly in the example was not only re-usability, but also avoiding having the long code for generating the data in the example itself. What about having it as it is now, in a file in the same folder as the examples. Would this be allowed?